### PR TITLE
20900-asNonTranslucentColor-is-misnamed-since-it-is-mutating-the-receiver-and-not-converting-it-

### PR DESCRIPTION
**Pull request by brackendev**
Fixes case #20900. https://pharo.fogbugz.com/f/cases/20900/asNonTranslucentColor-is-misnamed-since-it-is-mutating-the-receiver-and-not-converting-it

### CODE REVIEW

**Comment by astares** [comment]
You should deprecate the old Methode (instead of removing) and move it to deprecation package, ideally together with a transformation, so callers will be adopted.

**Comment by brackendev** [comment]
@astares Thank you.

Previous commit has been reverted. New commit has deprecation explanation and transformation.

Is there a good explanation (example or documentation) somewhere about deprecation packages?

**Comment by astares** [comment]
@brackendev 
Thanks. I wrote a "Deprecation guide" post on the mailinglist today so it hopefully is more clear on how to do it.

**Comment by brackendev** [comment]
@astares That post was extremely helpful (not only for this but for the future).

The latest commit moves #asNontranslucentColor to the Deprecated70 package.

**Comment by guillep** [comment]
retest this please

**Comment by guillep** [comment]
test this please

**Comment by guillep** [comment]
test this please

**Comment by guillep** [comment]
Sorry for the noise, I was trying to relaunch automatically the CI using comments/webhooks. But after more than one hour I failed. I'll relaunch it manually. Sorry again.

File: src/Colors/Color.class.st [hunk 1]
```diff
@@ -1711,6 +1704,13 @@ Color >> negated [
 		b: 1.0 - self blue
 ]
 
+{ #category : #conversions }
+Color >> nontranslucentColor [
```
Comment (jecisc) [inline src/Colors/Color.class.st]: Maybe `beOpaque` would be a more common name than translucent? But I don't know english enough to know what is the more common between the two.

I checked in an english dictionary online and there was no word `nontranslucent`. Maybe it should be `non translucent` and in that case the method should be called `nonTranslucentColor`?

**Review by jecisc — CHANGES_REQUESTED** [review]

File: src/Colors/Color.class.st (hunk 1 above)
Comment (brackendev) [reply src/Colors/Color.class.st]: Both good points, @jecisc. Cocoa's UIView has `isOpaque` (getter and setter), so `beOpaque` makes sense here. To fit in with others (i.e., `contrastingColor`, `dominantColor`), how about `opaqueColor`? @Ducasse, you opened the ticket, your thoughts?

Comment (brackendev) [reply src/Colors/Color.class.st]: Both good points, @jecisc. Cocoa's UIView has isOpaque (getter and setter), so beOpaque makes sense here. To fit in with others (i.e., contrastingColor, dominantColor), how about opaqueColor? If not, `nonTranslucentColor` works. @Ducasse, you opened the ticket, your thoughts?

Comment (brackendev) [reply src/Colors/Color.class.st]: Both good points, @jecisc. For example, Cocoa's UIView has `isOpaque` (getter and setter), so `beOpaque` could make sense here. To fit in with other methods in `conversions` (i.e., `contrastingColor`, `dominantColor`), how about `opaqueColor`? If not, `nonTranslucentColor` works. @Ducasse, you opened the ticket, your thoughts?

Comment (Ducasse) [reply src/Colors/Color.class.st]: I have no preference just that the previous name was misleading because it is about changing the receiver.

Comment (brackendev) [reply src/Colors/Color.class.st]: I renamed it to `opaqueColor` based on the reasoning above.

**Comment by jecisc** [comment]
I relaunched the CI.

**Comment by jecisc** [comment]
The crash of the CI seems related to the issue.

I think the PR should also update current users of the deprecated method.

File: src/Colors/Color.class.st (hunk 1 above)
Comment (Ducasse) [reply src/Colors/Color.class.st]: I will introduce beOpaque to convey the fact that this is modifying the receiver.
I will do it in another issue to see if the deprecation is not breaking the build

**Comment by Ducasse** [comment]
I will close this PR and the fogbugs issue and open a new one with just the rewriting of the user of asNonTranslucentColor

**Pull request closed without merging**